src/compute-pr-actions.ts

```diff
@@ -394,6 +379,34 @@ export function process(prInfo: PrInfo | BotEnsureRemovedFromProject | BotNoPack
     return context;
 }
 
+function makeStaleness(now: string, author: string, otherOwners: string[]) { // curried for convenience
+    return (kind: StalenessKind, since: Date,
+            freshDays: number, attnDays: number, nearDays: number,
+            doneColumn: ColumnName | "CLOSE") => {
+        const days = daysSince(since, now);
+        const state = days <= freshDays ? "fresh" : days <= attnDays ? "attention" : days <= nearDays ? "nearly" : "done";

```

Comment:
```suggestion
        if (days <= freshDays) return undefined;
        const state = days <= attnDays ? "attention" : days <= nearDays ? "nearly" : "done";
```
It looks like `{ kind: "fresh", ... }` and `undefined` are equivalent and the redundant state can be pruned? Just a thought.

Reply:
In theory, you're right -- and this is because a `fresh` state means that nothing is done, so it could have been dropped.

But in practice, there are two reasons I still kept it:

1. Currently, it's useful as a debugging tool: just looking at the `staleness` shows what timeline a PR is headed too, even if it has no implications ATM.  (And yes, there's no direct way currently to view this information, I'm working on a modification of the `daily` script that will show that too.)

2. In the future, the staleness could be used as another bit of information -- it just happens not to be useful right now.  For example,

   * `info.staleness?.kind === "YSYL"` can replace `info.canBeSelfMerged && info.mergeOfferDate`

   * `info.staleness?.kind === "Abandoned"` can replace `info.needsAuthorAction` (because it implies no `canBeSelfMerged`)

   (This could also happen with new staleness timelines too.)

   In any case, ATM, there's only the second one that could be done now.  I hesitate to do so, but I left a comment about it instead.  The reason I think this could be good in the future, is to avoid repeating information in a way that can lead to inconsistencies.  (Also, it's kind of a mental shift from a question of "does it need an action" to "is it on a path for abandonment".)

Reply:
I see. Replacing `needsAuthorAction` and `canBeSelfMerged` w/ `staleness` would make it more explicit that these states aren't indefinite. Seems good!